structs-in-rust/article.md

```diff
@@ -0,0 +1,366 @@
+// [3]: field value
+
+//               [1]       [2] [3]
+let p_unnamed = PointTuple(13, 37);
+
+// [1]: struct name
+// [2], [3]: field values
+```
+
+As you can see, the syntax of initialization is pretty straightforward. It is very similar to defining a JSON object, but with keys replaced with field names.
+
+Two things to keep in mind:
+- All fields must be assigned with values (there are no default parameters in Rust).
+- Order in which you list `key: value` pairs doesn't matter. You can do `let strange_order = Point { y: 37, x: 13 };` if you wish to.
+
+### Field access
+#### Get
+You can get the desired value using well-known dot notation.
+
+```
+let x = p_named.x;
+let y = p_named.y;
+```
+
+One may wonder, what is the syntax to address a specific field of a tuple struct. Fortunately, Rust has chosen the simplest way possible, by enumerating tuple components from *0* to *n*.
+
+```
+let x = p_unnamed.0;
+let y = p_unnamed.1;
+```
+#### Set
+It is possible to change the value of a field using the same dot notation, but the **struct instance must be defined as mutable**.
+```
+let mut p = PointTuple(1, 2);
+//  ^^^
+p.0 = 1000;
+assert_eq!(p.0, 1000);
+```
+Unlike some other languages, in Rust you cannot mark a specific field of a struct as mutable or immutable:
+
+```
+struct Struct {
+    mutable: mut i32,
+//           ^^^
+//           Expected type, found keyword `mut`.
+    immutable: bool,
+}
+```
+There are two options:
+- Have an immutable binding to the struct. Then you cannot change any of its fields.
+- Have a mutable binding, in which case you can set whatever field you choose.
+
+### Convenient field init syntax
+
+Let's define another struct and create its instance.
+
+```
+struct Bicycle {
+    brand: String,
+    kind: String,
+    size: u16,
+    suspension: bool,
+}
+
+let b1 = Bicycle {
+    brand: String::from("Brand A"),
+    kind: String::from("Mtb"),
+    size: 56,
+    suspension: true,
+};
+```
+
+#### Struct Update Syntax
+
+It often happens that you want to copy an instance of a struct and modify some (not all) of its values. Imagine that a different bicycle brand manufactures a model with identical parameters, and we need to create an instance of that model.
+
+Obviously, we can move all values manually:
+
+```
+let b2 = Bicycle {
+    brand: String::from("Other brand"),
+    kind: b1.kind,
+    size: b1.size,
+    suspension: b1.suspension,
+};
+```
+But this method involves too much code. **Struct update syntax** can help:
+```
+let b2 = Bicycle {
+    brand: String::from("Other brand"),
+    ..b1
+//  ^^ struct update syntax
+};
+```
+`..b1` tells Rust that we want to initialize the rest of `b2`'s fields with the values from `b1`.
+
+#### Field Init Shorthand
+It is possible to omit *field* in `field: value` initialization syntax if the value is a variable (or function argument) with a name that matches the field:
+```
+fn new_bicycle(brand: String) -> Bicycle {
+    let kind = String::from("road");
```

Comment:
I'd move this to the argument, it's simpler that way.

Reply:
Ok